Assets/Src/Scripts/Input/WebCamSource.cs

```diff
@@ -0,0 +1,158 @@
+using UnityEngine;
+using UnityEngine.UI;
+using Debug = UnityEngine.Debug;
+
+public class WebCamSource : MonoBehaviour {
+  public Dropdown CameraDropdown;
+  public Dropdown ResolutionDropdown;
+  public Dropdown FpsDropdown;
+  public Dropdown FlipDropdown;
+  public string DeviceName;
+  public int Fps = 30;
+  public bool Flip = false;
+  [SerializeField] private RenderTexture _buffer;
+  private WebCamTexture _webCam;
+  private int _width = 1280;
+  private int _height = 720;
+  private List<string> _devicesName = new List<string>();
+  private List<string> _resolutionOpt = new List<string>(){"1280X720","1600X900","1920X1080"};
+  private List<string> _fpsOpt = new List<string>(){"30Hz","60Hz"};
+  private List<string> _flipOpt = new List<string>(){"默认","水平翻转"};
+  //Start button event
+  public void Play() {
+    if(_webCam == null) {
+      _webCam = new WebCamTexture(DeviceName, _width, _height, Fps);
+      _webCam.Play();
+    }
+  }
+  //Stop button event
+  public void Stop() {
+    if(_webCam != null) {
+      _webCam.Stop();
+      _webCam = null;
+    }
+  }
+  //DropdownChange event
+  public void CameraDropdownChange(int value) {
+    //Debug.Log(_devicesName.ToArray()[value]);
+    if(_devicesName.ToArray()[value] != null) {
+      DeviceName = _devicesName.ToArray()[value];
+    }
+  }
+  //DropdownChange event
+  public void ResolutionDropdownChange(int value) {
+    string _resolution = _resolutionOpt.ToArray()[value];
+    if(_resolution != null) {
+      switch(_resolution) {
+        case "1280X720":
+          _width = 1280;
+          _height = 720;
+          break;
+        case "1600X900":
+          _width = 1600;
+          _height = 900;
+          break;
+        case "1920X1080":
+          _width = 1920;
+          _height = 1080;
+          break;
+        default:
+          break;
+      }
+      //Debug.Log(_width);
+    }
+  }
+  //DropdownChange event
+  public void FpsDropdownChange(int value) {
+    string _fps = _fpsOpt.ToArray()[value];
+    if(_fps != null) {
+      switch(_fps) {
+        case "60Hz":
+          Fps = 60;
+          break;
+        case "30Hz":
+          Fps = 30;
+          break;
+        default:
+          break;
+      }
+      //Debug.Log(Fps);
+    }
+  }
+  //DropdownChange event
+  public void FlipDropdownChange(int value) {
+    string _flip = _flipOpt.ToArray()[value];
+    if(_flip != null) {
+      switch(_flip) {
+        case "默认":
+          Flip = false;
+          break;
+        case "水平翻转":
+          Flip = true;
+          break;
+        default:
+          break;
+      }
+      //Debug.Log(Flip);
+    }
+  }
+  //Webcam and params init
+  IEnumerator Start() {
+    yield return Application.RequestUserAuthorization (UserAuthorization.WebCam);
+    if(Application.HasUserAuthorization(UserAuthorization.WebCam)) {
+      WebCamDevice[] _devices = WebCamTexture.devices;
+      CameraDropdown.ClearOptions();
+      for(int i = 0; i< _devices.Length; i++) {
```

Comment:
A
```suggestion
      for (int i = 0; i < _devices.Length; i++) {
```

A space is also required between "i" and "<".

And, this loop is recommended to be converted to a foreach loop, like:

```
foreach (var device in _device) {
  _devicesName.Add(device.name);
}
```